tests/test_format_detection.py

```diff
@@ -109,6 +109,34 @@ def detect(context):
         self.assertIn('*.txt', result.exception.failed_alternatives[0])
         self.assertIn('*.lst', result.exception.failed_alternatives[0])
 
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_require_files_success(self):
```

Comment:
I can't say it's a blocker, but consider splitting the detector tests into more specific ones. For example, this test currently covers at least 4 statements:
- sorting order of files
- exclusion of files
- confidence level on success 
- success itself

Reply:
I can only agree with regards to the 3rd bullet point - `test_empty_detector` already tests the default confidence level, so it's not necessary to check it in most other tests. I just added the asserts on the result value because I could. As for the rest:

* Success itself is impossible _not_ to test, since it's tested implicitly by virtue of having the call to `apply_format_detector` be unprotected by a `try` block.
* It's true that exclusion could be tested by a separate test. however, I didn't want to create a bunch of new tests, because the exclusion functionality is shared between `require_file` and `require_files` and so it's already tested by `require_file` tests. The reason I added an `exclude_fnames` parameter to the test at all is to make sure that the function forwards the value to the shared code.

Reply:
I removed the redundant confidence level checks.